src/main/java/com/metamx/common/StreamUtils.java

```diff
@@ -55,7 +55,9 @@ public static long copyToFileAndClose(InputStream is, File file) throws IOExcept
   {
     file.getParentFile().mkdirs();
     try (OutputStream os = new BufferedOutputStream(new FileOutputStream(file))) {
-      return ByteStreams.copy(is, os);
+      final long result =  ByteStreams.copy(is, os);
+      os.flush();
```

Comment:
the try with resources statement should flush and throw exceptions when closing the output stream. It shouldn't be necessary to flush here.